src/lib/operators/join_sort_merge/radix_cluster_sort.hpp

```diff
@@ -137,142 +129,138 @@ class RadixClusterSort {
-  * - manually select the clustering bits based on statistics.
-  * - consolidate clusters in order to reduce skew.
-  **/
-  std::unique_ptr<MaterializedSegmentList<T>> _radix_cluster(
-      std::unique_ptr<MaterializedSegmentList<T>>& input_chunks) {
+  // Performs least significant bit radix clustering which is used in the equi join case. Note: if we used the most
+  // significant bits, we could also use this for non-equi joins. Then, however we would have to deal with skewed
+  // clusters. Other ideas:
+  //    - manually select the clustering bits based on statistics.
+  //    - consolidate clusters in order to reduce skew.
+  MaterializedSegmentList<T> _radix_cluster(const MaterializedSegmentList<T>& input_chunks) {
     auto radix_bitmask = _cluster_count - 1;
     return _cluster(input_chunks, [=](const T& value) { return get_radix<T>(value, radix_bitmask); });
   }
 
-  /**
-  * Picks split values from the given sample values. Each split value denotes the inclusive
-  * upper bound of its corresponding cluster (i.e., split #0 is the upper bound of cluster #0).
-  * As the last cluster does not require an upper bound, the returned vector size is usually
-  * the cluster count minus one. However, it can be even shorter (e.g., attributes where
-  * #distinct values < #cluster count).
-  *
-  * Procedure: passed values are sorted and samples are picked from the whole sample
-  * value range in fixed widths. Repeated values are not removed before picking to handle
-  * skewed inputs. However, the final split values are unique. As a consequence, the split
-  * value vector might contain less values than `_cluster_count - 1`.
-  **/
-  const std::vector<T> _pick_split_values(std::vector<T> sample_values) const {
-    std::sort(sample_values.begin(), sample_values.end());
+  // Picks split values from the given sample values. Each split value denotes the inclusive upper bound of its
+  // corresponding cluster (i.e., split #0 is the upper bound of cluster #0). As the last cluster does not require an
+  // upper bound, the returned vector size is usually the cluster count minus one. However, it can be even shorter
+  // (e.g., attributes where #distinct values < #cluster count). Procedure: passed values are sorted and samples are
+  // picked from the whole sample value range in fixed widths. Repeated values are not removed before picking to handle
+  // skewed inputs. However, the final split values are unique. As a consequence, the split value vector might contain
+  // less values than `_cluster_count - 1`.
+  const std::vector<T> _pick_split_values(std::vector<T>& sample_values) const {
+    boost::sort::pdqsort(sample_values.begin(), sample_values.end());
 
     if (sample_values.size() <= _cluster_count - 1) {
       const auto last = std::unique(sample_values.begin(), sample_values.end());
       sample_values.erase(last, sample_values.end());
       return sample_values;
     }
```

Comment:
Well, sample_values is passed by value. That means the whole vector is copied. Code that uses this function do not see the changes. Correct?

Reply:
Am I wrong or was it passed by value and is now changed to be passed as `std::vector<T>& sample_values`?

Reply:
Don't know how it was before. But now that you bring it up, it's confusing. Better use by value and `std::move`?

Reply:
If I understand you correctly, the `std::move` shouldn't make a difference. Also, it might be fine to take `sample_values` by reference, I just wanted to bringt it up for discussion.

Reply:
Yes, shouldn't make a difference. I was rather thinking about intention. Since we simply pass the vector twice and both times don't use it afterwards, I thought whether passing it with `std::move()` might be the best way to signal just that.

Reply:
Using `std::move` and universal ref now.